lib/chef/provider/windows_task.rb

```diff
@@ -214,6 +217,35 @@ def task_need_update?
         false
       end
 
+      def compare_day
+        current_day = convert_system_dateformat_into_ruby_date(@current_resource.start_day).strftime("%m/%d/%Y")
+        new_day = convert_system_dateformat_into_ruby_date(new_resource.start_day).strftime("%m/%d/%Y")
+        current_day != new_day
+      end
+
+      def compare_time
+        time = convert_system_dateformat_into_ruby_date(@current_resource.start_time).strftime("%H:%M")
+        time != new_resource.start_time
+      end
+
+      def convert_system_dateformat_into_ruby_date(date_in_string)
+        Chef::Log.debug "Looking system date format"
+        task_script = <<-EOH
+          [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8
+          ([datetime]'#{date_in_string}').ToString('yyyy-MM-dd HH:mm')
+        EOH
+        date_time = powershell_out(task_script).stdout.force_encoding("UTF-8").gsub(/[\s+\uFEFF]/, "")
+        DateTime.strptime(date_time, "%Y-%m-%d %H:%M")
+      end
+
+      def convert_ruby_date_to_system_date
+        task_script = <<-EOH
+          [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8
+          ([datetime]"#{new_resource.start_day}").ToString([Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat.ShortDatePattern)
+        EOH
```

Comment:
Please add some comments to explain this code. Comments can be added here or at the place where this method is being called.